### ISSUE TITLE
Web extension contextmenu API

### ISSUE THREAD

**Issue by ahmadayubi** [Maps4HTML/mapml-extension#5]
Web extensions allow developers to define their own custom context menu items. This maybe useful for MapML but may not be required.

**Comment by prushforth** [comment]
Do you mean to customise the browser's context menu?  That might be interesting, although I wouldn't want to separate the context menu functionality out of our custom elements and put it only into the extension.

**Comment by ahmadayubi** [comment]
Yeah I meant it allows the ability to add to the browser's context menu, if this is ever implemented you'd have to completely get rid of the context menu inside MapML, as you couldn't have both. Although you could conditionally show different ones (either the browser's context menu or MapML's context menu) depending on if the extension is installed or not.
